File: app/src/main/java/androidx/core/app/SafeJobIntentService.java
package androidx.core.app;

import info.metadude.android.eventfahrplan.commons.logging.Logging;

/**
 * Mitigates the effect of a SecurityException which can occur in {@link JobIntentService}.
 * The issue is tracked here: https://issuetracker.google.com/issues/63622293
 */
public abstract class SafeJobIntentService extends JobIntentService {

    private static final String LOG_TAG = "SafeJobIntentService";

    @Override
    GenericWorkItem dequeueWork() {
        try {
            return super.dequeueWork();
        } catch (SecurityException e) {
            // There is not much we can do here.
            Logging.get().report(LOG_TAG, "" + e.getMessage());
        }
        return null;
    }

}

File: app/src/main/java/androidx/core/app/SafeJobIntentService.kt
package androidx.core.app

import info.metadude.android.eventfahrplan.commons.logging.Logging

/**
 * Mitigates the effect of a SecurityException which can occur in [JobIntentService].
 * The issue is tracked here: https://issuetracker.google.com/issues/63622293
 */
abstract class SafeJobIntentService : JobIntentService() {

    private companion object {
        const val LOG_TAG = "SafeJobIntentService"
    }

    private val logging = Logging.get()

    internal override fun dequeueWork() =
        try {
            super.dequeueWork()
        } catch (e: SecurityException) {
            // There is not much we can do here.
            logging.report(LOG_TAG, "$e")
            null
        }

}
